trie/hasher.go

```diff
@@ -121,17 +135,44 @@ func (h *hasher) hashChildren(original node, db DatabaseWriter) (node, node, err
 
 	case *fullNode:
 		// Hash the full node's children, caching the newly hashed subtrees
+		var wg sync.WaitGroup
```

Comment:
You only use this WaitGroup in a deeper nesting. Please move it inside, there's no need to declare it out here.